File: app/src/main/java/com/usher/demo/main/DemoConfig.java
package com.usher.demo.main;

import com.usher.demo.awesome.channel.ChannelActivity;
import com.usher.demo.awesome.decoration.DecorationActivity;
import com.usher.demo.awesome.drag.DragActiity;
import com.usher.demo.awesome.selection.SelectionActivity;
import com.usher.demo.awesome.smarthome.SmartHomeActivity;
import com.usher.demo.image.BlurActivity;
import com.usher.demo.image.RoundImageActivity;
import com.usher.demo.kotlin.KotlinActivity;
import com.usher.demo.material.FitsSystemWindowActivity;
import com.usher.demo.material.ProfileActivity;
import com.usher.demo.material.TabLayoutActivity;
import com.usher.demo.material.home.HomeActivity;
import com.usher.demo.other.launchmode.AActivity;
import com.usher.demo.other.notification.NotificationActivity;
import com.usher.demo.rx.RxExitActivity;
import com.usher.demo.rx.RxSearchActivity;
import com.usher.demo.rx.RxSplashActivity;
import com.usher.demo.rx.RxSumActivity;
import com.usher.demo.view.ChartActivity;
import com.usher.demo.view.CircleWaveActivity;
import com.usher.demo.view.MarqueeTextActivity;
import com.usher.demo.view.WaveActivity;
import com.usher.demo.view.loading.LoadingActivity;
import com.usher.demo.view.looprecycler.LoopRecyclerActivity;
import com.usher.demo.view.pager.PagerActivity;
import com.usher.demo.web.AngularActivity;
import com.usher.demo.web.three.ThreeActivity;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class DemoConfig {
    private static final Map<String, List<DemoItem>> configMap = new HashMap<>();
    static final String TAG_KEY = "TAG_KEY";
    private static final String KEY_RX = "rx";
    private static final String KEY_WEB = "web";
    private static final String KEY_IMAGE = "image";
    private static final String KEY_VIEW = "view";
    private static final String KEY_MATERIAL = "material";
    private static final String KEY_AWESOME = "awesome";
    private static final String KEY_OTHER = "other";
    private static final String KEY_KOTLIN = "kotlin";

    static {

        configMap.put("main", Arrays.asList(
                new DemoItem("rx", KEY_RX, DemoListActivity.class),
                new DemoItem("image", KEY_IMAGE, DemoListActivity.class),
                new DemoItem("material", KEY_MATERIAL, DemoListActivity.class),
                new DemoItem("view", KEY_VIEW, DemoListActivity.class),
                new DemoItem("web", KEY_WEB, DemoListActivity.class),
                new DemoItem("other", KEY_OTHER, DemoListActivity.class),
                new DemoItem("awesome", KEY_AWESOME, DemoListActivity.class),
                new DemoItem("kotlin", KEY_KOTLIN, KotlinActivity.class)
        ));

        configMap.put(KEY_RX, Arrays.asList(
                new DemoItem("splash", RxSplashActivity.class),
                new DemoItem("sum", RxSumActivity.class),
                new DemoItem("exit", RxExitActivity.class),
                new DemoItem("search", RxSearchActivity.class)
        ));

        configMap.put(KEY_IMAGE, Arrays.asList(
                new DemoItem("blur", BlurActivity.class),
                new DemoItem("picasso transform", RoundImageActivity.class)
        ));

        configMap.put(KEY_WEB, Arrays.asList(
                new DemoItem("three", ThreeActivity.class),
                new DemoItem("angular", AngularActivity.class)
        ));

        configMap.put(KEY_MATERIAL, Arrays.asList(
                new DemoItem("fitsSystemWindow", FitsSystemWindowActivity.class),
                new DemoItem("tab", TabLayoutActivity.class),
                new DemoItem("profile", ProfileActivity.class),
                new DemoItem("home", HomeActivity.class)

        ));

        configMap.put(KEY_VIEW, Arrays.asList(
                new DemoItem("wave", WaveActivity.class),
                new DemoItem("pager", PagerActivity.class),
                new DemoItem("marquee text", MarqueeTextActivity.class),
                new DemoItem("loading", LoadingActivity.class),
                new DemoItem("chart", ChartActivity.class),
                new DemoItem("loop reycler", LoopRecyclerActivity.class),
                new DemoItem("circle wave", CircleWaveActivity.class)
        ));

        configMap.put(KEY_OTHER, Arrays.asList(
                new DemoItem("notification", NotificationActivity.class),
                new DemoItem("launch mode", AActivity.class)
        ));

        configMap.put(KEY_AWESOME, Arrays.asList(
                new DemoItem("selection", SelectionActivity.class),
                new DemoItem("edit channel", ChannelActivity.class),
                new DemoItem("recyclerView decoration", DecorationActivity.class),
                new DemoItem("smart home", SmartHomeActivity.class),
                new DemoItem("drag", DragActiity.class)
        ));
    }

    static List<DemoItem> getDemoConfig(String key) {
        return configMap.get(key);
    }
}

File: app/src/main/java/com/usher/demo/main/DemoConfig.kt
package com.usher.demo.main

import com.usher.demo.awesome.LoopHintActivity
import com.usher.demo.awesome.ZipCodeActivity
import com.usher.demo.awesome.decoration.ContactsActivity
import com.usher.demo.awesome.decoration.StickyHeaderActivity
import com.usher.demo.awesome.drag.DragActiity
import com.usher.demo.awesome.itemtouch.ChannelEditActivity
import com.usher.demo.demonstration.*
import com.usher.demo.material.FitsSystemWindowActivity
import com.usher.demo.material.ProfileActivity
import com.usher.demo.material.TabLayoutActivity
import com.usher.demo.material.home.HomeActivity
import com.usher.demo.other.LogActivity
import com.usher.demo.other.launchmode.AActivity
import com.usher.demo.other.notification.NotificationActivity
import com.usher.demo.rx.RxExitActivity
import com.usher.demo.rx.RxSearchActivity
import com.usher.demo.rx.RxSplashActivity
import com.usher.demo.rx.RxSumActivity
import com.usher.demo.view.*
import com.usher.demo.view.loading.LoadingActivity
import com.usher.demo.view.seat.SeatSelection2Activity
import com.usher.demo.view.seat.SeatSelectionActivity
import com.usher.demo.web.AngularActivity
import com.usher.demo.web.three.ThreeActivity

object DemoConfig {
    private val configMap = hashMapOf<String, List<DemoItem>>()
    const val TAG_KEY = "TAG_KEY"
    private const val KEY_MAIN = "main"
    private const val KEY_RX = "rx"
    private const val KEY_WEB = "web"
    private const val KEY_DEMONSTRATION = "demonstration"
    private const val KEY_VIEW = "view"
    private const val KEY_MATERIAL = "material"
    private const val KEY_AWESOME = "awesome"
    private const val KEY_OTHER = "other"

    init {
        configMap[KEY_MAIN] = listOf(
                DemoItem(KEY_RX, DemoListActivity::class.java, "RX"),
                DemoItem(KEY_DEMONSTRATION, DemoListActivity::class.java, "Demonstration"),
                DemoItem(KEY_MATERIAL, DemoListActivity::class.java, "Material"),
                DemoItem(KEY_VIEW, DemoListActivity::class.java, "View"),
                DemoItem(KEY_WEB, DemoListActivity::class.java, "Web"),
                DemoItem(KEY_OTHER, DemoListActivity::class.java, "Other"),
                DemoItem(KEY_AWESOME, DemoListActivity::class.java, "Awesome")
        )

        configMap[KEY_RX] = listOf(
                DemoItem(RxSplashActivity::class.java, "Splash"),
                DemoItem(RxSumActivity::class.java, "Sum"),
                DemoItem(RxExitActivity::class.java, "Exit"),
                DemoItem(RxSearchActivity::class.java, "Search")
        )

        configMap[KEY_DEMONSTRATION] = listOf(
                DemoItem(ImageScaleTypeActivity::class.java, "ScaleType"),
                DemoItem(BitmapXferActivity::class.java, "Bitmap Xfer"),
                DemoItem(BitmapBlurActivity::class.java, "Bitmap Blur", "RenderScript"),
                DemoItem(PorterDuffColorFilterActivity::class.java, "PorterDuff", "ColorFilter"),
                DemoItem(LightingColorFilterActivity::class.java, "Lighting", "ColorFilter"),
                DemoItem(PorterDuffXfermodeActivity::class.java, "PorterDuff", "Xfermode"),
                DemoItem(BitmapMeshWarpActivity::class.java, "Warp", "Bitmap Mesh"),
                DemoItem(BitmapMeshRippleActivity::class.java, "Ripple", "Bitmap Mesh"),
                DemoItem(BitmapMeshCurtainActivity::class.java, "Curtain", "Bitmap Mesh"),
                DemoItem(ViewPager2Activity::class.java, "ViewPager2"),
                DemoItem(TabLayout1Activity::class.java, "TabLayout", "ViewPager"),
                DemoItem(TabLayout2Activity::class.java, "TabLayout", "ViewPager2")
        )

        configMap[KEY_WEB] = listOf(
                DemoItem(ThreeActivity::class.java, "three"),
                DemoItem(AngularActivity::class.java, "angular")
        )

        configMap[KEY_MATERIAL] = listOf(
                DemoItem(FitsSystemWindowActivity::class.java, "fitsSystemWindow"),
                DemoItem(TabLayoutActivity::class.java, "tab"),
                DemoItem(ProfileActivity::class.java, "profile"),
                DemoItem(HomeActivity::class.java, "home")
        )

        configMap[KEY_VIEW] = listOf(
                DemoItem(MarqueeTextActivity::class.java, "Marquee Text"),
                DemoItem(WaveActivity::class.java, "Wave"),
                DemoItem(LoopPagerActivity::class.java, "Loop Pager"),
                DemoItem(LoopPager2Activity::class.java, "Loop Pager2"),
                DemoItem(SeatSelectionActivity::class.java, "Seat Selection"),
                DemoItem(SeatSelection2Activity::class.java, "Seat Selection2"),
                DemoItem(LoadingActivity::class.java, "Loading"),
                DemoItem(ChartActivity::class.java, "Chart"),
                DemoItem(LoopRecyclerActivity::class.java, "Loop Recycler"),
                DemoItem(ScanActivity::class.java, "Scan"),
                DemoItem(IndexActivity::class.java, "Index"),
                DemoItem(ColorPickerActivity::class.java, "Color Picker"),
                DemoItem(ColorSeekerActivity::class.java, "Color Seeker"),
                DemoItem(CurtainActivity::class.java, "Curtain"),
                DemoItem(GraffitiActivity::class.java, "Graffiti")
        )

        configMap[KEY_OTHER] = listOf(
                DemoItem(NotificationActivity::class.java, "notification"),
                DemoItem(AActivity::class.java, "launch mode"),
                DemoItem(LogActivity::class.java, "log")
        )

        configMap[KEY_AWESOME] = listOf(
                DemoItem(ChannelEditActivity::class.java, "Channel Edit"),
                DemoItem(StickyHeaderActivity::class.java, "Sticky Header"),
                DemoItem(ContactsActivity::class.java, "Contacts"),
                DemoItem(LoopHintActivity::class.java, "Loop Hint"),
                DemoItem(ZipCodeActivity::class.java, "ZipCode Input"),
                DemoItem(DragActiity::class.java, "Drag")
        )
    }

    fun getDemoConfig(key: String?): List<DemoItem> = configMap[key ?: KEY_MAIN] ?: listOf()
}

data class DemoItem(val key: String?, val cls: Class<*>, val title: String, val desc: String?) {
    constructor(key: String, cls: Class<*>, title: String) : this(key, cls, title, null)
    constructor(cls: Class<*>, title: String) : this(null, cls, title, null)
    constructor(cls: Class<*>, title: String, desc: String) : this(null, cls, title, desc)
}